app/build.gradle.kts

```diff
@@ -1,196 +1,36 @@
-    }
-
-    testOptions {
-        unitTests.isReturnDefaultValues = true
-    }
-
-    tasks.withType<Test>().configureEach {
-        useJUnitPlatform {
-            includeEngines("spek2")
-        }
-    }
-
-    lint {
-        abortOnError = false
-        disable += "MissingTranslation"
-    }
 }
 
 dependencies {
-    implementation(project(":common"))
-
-    coreLibraryDesugaring(libs.tools.desugar.jdk)
-
-    implementation(libs.blurView)
-    "fullImplementation"(libs.androidx.health.connect.client)
-
-    implementation(libs.kotlin.stdlib)
-    implementation(libs.kotlin.reflect)
-    implementation(libs.kotlinx.coroutines.core)
-    implementation(libs.kotlinx.coroutines.android)
-    "fullImplementation"(libs.kotlinx.coroutines.play.services)
-
-    implementation(libs.hilt.android)
-    ksp(libs.hilt.android.compiler)
-
-    implementation(libs.appcompat)
-    implementation(libs.androidx.lifecycle.runtime.ktx)
-    implementation(libs.constraintlayout)
-    implementation(libs.recyclerview)
-    implementation(libs.preference.ktx)
-    implementation(libs.material)
-    implementation(libs.fragment.ktx)
-
-    implementation(libs.jackson.module.kotlin)
-    implementation(libs.okhttp)
-
-    implementation(libs.bundles.coil)
-
-    "fullImplementation"(libs.play.services.location)
-    "fullImplementation"(libs.play.services.home)
-    "fullImplementation"(libs.play.services.threadnetwork)
-    "fullImplementation"(platform(libs.firebase.bom))
-    "fullImplementation"(libs.firebase.messaging)
-    "fullImplementation"(libs.sentry.android)
-    "fullImplementation"(libs.play.services.wearable)
-    "fullImplementation"(libs.wear.remote.interactions)
-
-    implementation(libs.biometric)
-    implementation(libs.webkit)
-
-    implementation(libs.bundles.media3)
-    "fullImplementation"(libs.media3.datasource.cronet)
-    "minimalImplementation"(libs.media3.datasource.cronet) {
-        exclude(group = "com.google.android.gms", module = "play-services-cronet")
-    }
-    "minimalImplementation"(libs.cronet.embedded)
-
-    implementation(platform(libs.compose.bom))
-    implementation(libs.compose.animation)
-    implementation(libs.compose.foundation)
-    implementation(libs.compose.material)
-    implementation(libs.compose.material.icons.core)
-    implementation(libs.compose.material.icons.extended)
-    implementation(libs.compose.runtime)
-    implementation(libs.compose.ui)
-    implementation(libs.compose.uiTooling)
-    implementation(libs.activity.compose)
-    implementation(libs.navigation.compose)
-    implementation(libs.androidx.lifecycle.runtime.compose)
-
-    implementation(libs.iconics.core)
-    implementation(libs.iconics.compose)
-    implementation(libs.community.material.typeface)
-
-    implementation(libs.bundles.paging)
-
-    implementation(libs.reorderable)
-    implementation(libs.changeLog)
-
-    implementation(libs.zxing)
-    implementation(libs.improv)
-
-    implementation(libs.car.core)
```

Comment:
IMO it was a source of error to have a duplicated list of dependencies. Someone could have missed to add a deps into `:automotive` while adding it to `:app`. 

My 2ct on this is that we might want to consider another approach for this and avoid using the source of `:app` in another module which btw raise a warning about duplicates in Android Studio.

Reply:
Not saying you're wrong (none of us are Gradle experts), and it always felt a little clunky to me, but this was originally contributed by someone from Google's Automotive team.

Missing a dependency is a risk but there is a review / it probably would break when building.

Reply:
Depending on the deps that is missing, it could manage to build and fail at runtime. I'm thinking about something that would use reflection (even if I think we are not using it here).

If we were to go the path of removing this source sharing, we would need to create another layer with a gradle module and extract some part of `:app` into it. It would allow to have specific code for `automotive` isolated from the rest. WDYT? Could be useful to if we want to have a dedicated experience for AndroidTV.